jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -952,7 +952,10 @@ public void demand(Runnable demandCallback)
                 if (!error)
                 {
                     if (httpChannelState._onContentAvailable != null)
+                    {
+                        LOG.warn("demand pending state={} stream={}", httpChannelState, stream);
```

Comment:
The stream classes have no `toString()` so `stream` can be omitted as it won't tell us much.

Instead of a `LOG.warn()` maybe it would be better to include the `httpChannelState`'s string representation in the exception's message itself, like the following?

```java
throw new IllegalArgumentException("demand pending on " + httpChannelState);
```